package m_Chapter12;

public class B_Exception {
	public static void main(String[] args) {
		//! ReuntimeException
		// : 프로그르래머의 실수에 의해 발생하는 에외들을 포함
		// : 컴파일 시점에서 체크되지 X (unchecked exception)
		// NullPointerException, IndexOutOfBoundsExcpetion 등
		
		// 1. NullPointerException 
		// : 참조변수기 null 값을 가르킬 떄, 해당 변수를 사용할 경우 발생
		try {
			String text = null;
			int length = text.length();
		} catch (NullPointerException e) {
			System.out.println("Null 참조 에러");
		}
		
		// 2. ArrayIndexOutOfBoundsExcpetion
		// : 배열의 범위를 초과하는 인덱스에 접근할 때 발생
		
		int[] array = new int[10]; // 배열의 길이 10 (인덱스 0~9)
		
		try {
			array[10] = 5;
		} catch (ArrayIndexOutOfBoundsException e) {
			System.out.println("배열 범위 초과");
		}
		
		// 자바의 기타 예외 클래스
		// IOException
		// : 입출력 작업 중 예외가 발생할 경우 사용되는 예외
		// : checked exception- 컴파일 시점에서
	}
}
